### ISSUE TITLE
[VEDA2 Refactor] Establish Pattern and Remove Routing Dependencies from Data Catalog Feature

### ISSUE THREAD

**Issue by sandrahoang686** [NASA-IMPACT/veda-ui#1108]

**Comment by hanbyul-here** [comment]
From Data Catalog implementation try, we could find the routing problems that need to be handled.

### Base finding: react-router-dom component doesn't work with Next instance, and Next has its own Link component.

1. `<NavLink>` of `react-router-dom` embedded in source code
  One of the possible solutions is to Pass Link Components from Consumer level
2. Components using  react-router for their behaviors (ex. sticky header using `useLocation` of `react-router-dom` to detect page transition )
  We should first identify these components and approach them holistically. ex. Do we need this behavior when it is not used in SPA environment? 
3. Query parameter manager needing `react-router-dom` (`useQSState hook` requires `navigate` from `react-router-dom`'s `useNavigate` as a parameter)
  One of the possible solutions is to make this as a prop

4. Make sure the link works with routings defined on the instance level. (ex. does dataset catalog card work with different DATACATALOG_PATH ?)

**Comment by sandrahoang686** [comment]
_Update_: This has been kindof been worked on when during breakout components by having the Link Component pass down as a prop. But there are some hacks in there that need to be clean up and we should relook at how we are doing this to see if it is the most optimal and clean... 

Going to think about what we should do with this ticket...

**Comment by vgeorge** [comment]
I will have limited availability in the next few weeks and wanted to provide a status update in case someone else has time to tackle the remaining work.

In #1270 and #1276, we reduced the dependency on React Router. I believe the next steps are:

1. **Remove usage of the `useLocation` hook**:  
   This should be straightforward since most occurrences involve reading `location.pathname`. These can be replaced with the new custom `usePathname` hook.

2. **Refactor usage of linkProperties**:  
   I suggest focusing on making the components independent of routing and navigation logic, as proposed by @sandra. Adding a context provider might complicate usage in Next.js, so I'd avoid it.

   I'd recommend addressing this interactively at the leaf component level to prevent unintended breaks in different parts of the app. For example, the Catalog Card is a good starting point. We could introduce an `href` prop that, when provided, makes the card act as a link and ignores `onClick` events. We would still need a `Link` element to support both Next.js and SPAs. Below is a sample implementation of a `CustomLink` component:

```tsx
   import React from 'react';

  // try loading dependencies during build time
   let NextLink: any;
   try {
     NextLink = require('next/link').default;
   } catch (e) {
     NextLink = null;
   }

   let ReactRouterLink: any;
   try {
     ReactRouterLink = require('react-router-dom').Link;
   } catch (e) {
     ReactRouterLink = null;
   }

   interface CustomLinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
     to?: string;
     href?: string;
     children: React.ReactNode;
   }

   export const CustomLink = ({ to, href, children, ...props }: CustomLinkProps) => {
     const path = to || href;

     if (NextLink) {
       return (
         <NextLink href={path} {...props}>
           {children}
         </NextLink>
       );
     }

     if (ReactRouterLink) {
       return (
         <ReactRouterLink to={path} {...props}>
           {children}
         </ReactRouterLink>
       );
     }

     // Fallback to a regular anchor tag if no routing library is available
     return (
       <a href={path} {...props}>
         {children}
       </a>
     );
   };
```

**Comment by sandrahoang686** [comment]
Thanks for the input & thoughts @vgeorge 🙇‍♀ . I'm thinking if we are going to tackle the issue with routing coupled up with the components we should probably fully tackle it since this ticket was meant for that. I'm thinking the proposed solution above might still be some sort of bandage as it still worries about "how to navigate" in our veda-ui components when it'd be ideal to totally move away from that. But I agree with starting with the card component to iterate on removing routing entirely from our library. Another thing is, we are using Next in our template instance but ideally we dont want the veda-ui library to care or know - if someone wanted to use our library with Gatsby, they should be able too (I think gatsby works with react-router-dom though but has its own preferred routing mechanism). Thinking that creating some type custom routing controller component that handles based on framework isn't too scalable and is hence more bandagey?? I'll experiment with a first pass at removing routing in one of our components to see how that goes and we could bounce off ideas from that PR :) 
PR experimentation here: https://github.com/NASA-IMPACT/veda-ui/pull/1310/files

**Comment by sandrahoang686** [comment]
This will be done in iterations as we work to remove components in veda-ui from being tightly integrated to routing/nav.
This can close after PR #1310  is merged. 

Follow up tickets have been created here...
* Remove LinkProperties from StoriesHub Component https://github.com/NASA-IMPACT/veda-ui/issues/1325
* Remove LinkProperties from EA Component https://github.com/NASA-IMPACT/veda-ui/issues/1326
* Remove LinkProperties from Card Component [PLACEHOLDER] https://github.com/NASA-IMPACT/veda-ui/issues/1327

**Comment by vgeorge** [comment]
@sandrahoang686 while I agree we shouldn't be handling every framework use case, I don't think relying only on event handlers like `onDatasetClick` is the ideal approach. This will make the components stop working as semantic links, affecting accessibility. One possible approach is to allow the parent component to pass an `<a>`, a Next.js `<Link>` or whatever element in a `as` property, a common approach on many component libraries. Let's follow up on this in #1327.

**Comment by sandrahoang686** [comment]
@vgeorge good point, i should've been more clear. I agree - I think callbacks are good for actions so like buttons for example. This is why I did not include the `PageHeader` as part of this work because it should retain the link components.
